app/lib/foreman_deployments/tasks/search_task_definition.rb

```diff
@@ -7,6 +7,26 @@ def run
           results = SerializableArray.new(results)
           SearchTaskDefinition.create_output(results, output)
         end
+
+        def self.dereference_output(storage, path)
```

Comment:
----- Original Message -----
> > @@ -7,6 +7,26 @@ def run
> >            results = SerializableArray.new(results)
> >            SearchTaskDefinition.create_output(results, output)
> >          end
> > +
> > +        def self.dereference_output(storage, path)
> 
> OK, the use case with a condition that I have described is indeed very
> advanced. We can disallow it (at least for now), but what about the original
> problem that I describe in (2)?
> 
> > I see an unusual task ordering here - plan envelope task -> plan
> > CreateHostTask -> execute CreateHostTask-> plan DhcpReserve task ->
> > execute DhcpReserveTask

No, it's still `plan CreateHost -> plan DhcpReserve -> execute CreateHost -> execute DhcpReserve`
It's not intended to do much with the inputs in the plan phase: the plan phase
is really mainly for planning and validation if possible.

> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/theforeman/foreman_deployments/pull/24/files#r55202579